File: InPlan.Web/Modules/Parametros/Proyectos/ProyectosRow.cs
using InPlan.Tenencia;
using Serenity.ComponentModel;
using Serenity.Data;
using Serenity.Data.Mapping;
using System;
using System.ComponentModel;

namespace InPlan.Parametros;

[ConnectionKey("Default"), Module("Parametros"), TableName("Proyectos")]
[DisplayName("Proyectos"), InstanceName("Proyecto")]
[ReadPermission("Administration:General")]
[ModifyPermission("Administration:General")]
[LookupScript]
public sealed partial class ProyectosRow : Row<ProyectosRow.RowFields>, IIdRow, INameRow
{
    const string jIdEmpresa = nameof(jIdEmpresa);
    const string jIdArea = nameof(jIdArea);

    class RowTemplate
    {
        [DisplayName("Id Proyecto"), Identity, IdProperty]
        public int? IdProyecto { get; set; }

        [LookupEditor(typeof(EmpresasRow))]
        [DisplayName("Empresa"), NotNull, ForeignKey("Empresas", "IdEmpresa"), LeftJoin(jIdEmpresa), TextualField(nameof(IdEmpresaDescripcionCorta))]
        public int? IdEmpresa { get; set; }

        [LookupEditor(typeof(AreasRow))]
        [DisplayName("Área"), NotNull, ForeignKey("Areas", "IdArea"), LeftJoin(jIdArea), TextualField(nameof(IdAreaDescripcion))]
        public int? IdArea { get; set; }

        [DisplayName("Nombre"), Size(250), NotNull, QuickSearch, NameProperty]
        public string NombreProyecto { get; set; }

        [DisplayName("Descripción"), Size(250)]
        public string DescripcionProyecto { get; set; }

        [DisplayName("Nro Ticket"), Size(50)]
        public string NroTicket { get; set; }

        [DisplayName("Fecha Carga"), NotNull]
        public DateTime? FechaCarga { get; set; }

        [DisplayName("Fecha Inicio"), NotNull]
        public DateTime? FechaInicio { get; set; }

        [DisplayName("Fecha Fin")]
        public DateTime? FechaFin { get; set; }

        [DisplayName("Documentos Relacionados"), MultipleFileUploadEditor]
        public string DocumentosRelacionados { get; set; }

        [DisplayName("Observaciones")]
        public string Observaciones { get; set; }

        [DisplayName("Baja"), NotNull]
        public bool? Baja { get; set; }

        [DisplayName("Empresa"), Expression($"{jIdEmpresa}.[DescripcionCorta]")]
        public string IdEmpresaDescripcionCorta { get; set; }

        [DisplayName("Área"), Expression($"{jIdArea}.[Descripcion]")]
        public string IdAreaDescripcion { get; set; }
    }
}

File: InPlan.Web/Modules/ServerTypes/Parametros/ProyectosRow.ts
import { getLookup, getLookupAsync, fieldsProxy } from "@serenity-is/corelib/q";

export interface ProyectosRow {
    IdProyecto?: number;
    IdEmpresa?: number;
    IdArea?: number;
    NombreProyecto?: string;
    DescripcionProyecto?: string;
    NroTicket?: string;
    FechaCarga?: string;
    FechaInicio?: string;
    FechaFin?: string;
    DocumentosRelacionados?: string;
    Observaciones?: string;
    Baja?: boolean;
    IdEmpresaDescripcionCorta?: string;
    IdAreaDescripcion?: string;
}

export abstract class ProyectosRow {
    static readonly idProperty = 'IdProyecto';
    static readonly nameProperty = 'NombreProyecto';
    static readonly localTextPrefix = 'Parametros.Proyectos';
    static readonly lookupKey = 'Parametros.Proyectos';

    /** @deprecated use getLookupAsync instead */
    static getLookup() { return getLookup<ProyectosRow>('Parametros.Proyectos') }
    static async getLookupAsync() { return getLookupAsync<ProyectosRow>('Parametros.Proyectos') }

    static readonly deletePermission = 'Administration:General';
    static readonly insertPermission = 'Administration:General';
    static readonly readPermission = 'Administration:General';
    static readonly updatePermission = 'Administration:General';

    static readonly Fields = fieldsProxy<ProyectosRow>();
}